[WebDriver][BiDi] bad_variant_access in BidiScriptAgent::callFunction if exception is raised

lauromoura opened the pull request:
#### 890d36fb09e9ca48ba9655581e98b43fb58cb01a
<pre>
[WebDriver][BiDi] bad_variant_access in BidiScriptAgent::callFunction if exception is raised
<a href="https://bugs.webkit.org/show_bug.cgi?id=295890">https://bugs.webkit.org/show_bug.cgi?id=295890</a>

Reviewed by BJ Burg.

Only parse the result value if we got a successful evaluation.

This commit also adds basic support to extract a JavascriptError (i.e.
exception) information inside an ExceptionDetail, leaving the ASYNC_FAIL
family of macros for other non-JS errors.

Note that we&apos;d still likely need to replace/extend `evaluateJSFunction` to
extract the exception information instead of just the string value in
the return.

* Source/WebKit/UIProcess/Automation/BidiScriptAgent.cpp:
(WebKit::BidiScriptAgent::callFunction):

Canonical link: <a href="https://commits.webkit.org/299146@main">https://commits.webkit.org/299146@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9603598e8ff81c2eaa4e1f1275c12e18a3f0dc13

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37701 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28337 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124173 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70058 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38394 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46283 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89562 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120975 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30579 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105829 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70054 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29647 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67836 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100002 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24124 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127254 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44926 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33851 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98238 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45287 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102053 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98024 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24931 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43420 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21408 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41361 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44798 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50472 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44258 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47603 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45947 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/91edc7d33dde286e06110f7a81005384ddae16b1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48013

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9603598e8ff81c2eaa4e1f1275c12e18a3f0dc13. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48013

lauromoura committed:
[WebDriver][BiDi] bad_variant_access in BidiScriptAgent::callFunction if exception is raised
https://bugs.webkit.org/show_bug.cgi?id=295890

Reviewed by BJ Burg.

Only parse the result value if we got a successful evaluation.

This commit also adds basic support to extract a JavascriptError (i.e.
exception) information inside an ExceptionDetail, leaving the ASYNC_FAIL
family of macros for other non-JS errors.

Note that we'd still likely need to replace/extend `evaluateJSFunction` to
extract the exception information instead of just the string value in
the return.

* Source/WebKit/UIProcess/Automation/BidiScriptAgent.cpp:
(WebKit::BidiScriptAgent::callFunction):

Canonical link: https://commits.webkit.org/299146@main

webkit-commit-queue commented:
Committed 299146@main (890d36fb09e9): <https://commits.webkit.org/299146@main>

Reviewed commits have been landed. Closing PR #48013 and removing active labels.
